sncosmo/models.py

```diff
@@ -573,6 +573,96 @@ def _flux(self, phase, wave):
+
+    Parameters
+    ----------
+    modeldir : str, optional
+        Directory path containing model component files. Default is `None`,
+        which means that no directory is prepended to filenames when
+        determining their path.
+        m0file, alpha1file, alpha2file, alpha3file, CCMfile: str or fileobj, optional
+        Filenames of various model components. Defaults are:
+
+        * m0file = 'sugar_template_0.dat' (2-d grid)
+        * alpha1file = 'sugar_template_1.dat' (2-d grid)
+        * alpha2file = 'sugar_template_2.dat' (2-d grid)
+        * alpha3file = 'sugar_template_3.dat' (2-d grid)
+        * CCMfile = 'sugar_template_4.dat' (2-d grid)
+
+    Notes
+    -----
+    The "2-d grid" files have the format ``<phase> <wavelength>
+    <value>`` on each line.
+    """
+    _param_names = ['Xgr', 'q1', 'q2', 'q3', 'A']
+    param_names_latex = ['X_r', 'q_1', 'q_2', 'q_3', 'A']
+
+    def __init__(self, modeldir=None,
+                 m0file='sugar_template_0.dat',
+                 alpha1file='sugar_template_1.dat',
+                 alpha2file='sugar_template_2.dat',
+                 alpha3file='sugar_template_3.dat',
+                 CCMfile='sugar_template_4.dat',
+                 name=None, version=None):
+
+        self.name = name
+        self.version = version
+        self._model = {}
+        self._parameters = np.array([1e-15, 1., 1., 1., 0.])
+        self.M_keys = ['M0', 'ALPHA1', 'ALPHA2', 'ALPHA3', 'CCM']
+        names_or_objs = {'M0': m0file,
+                         'ALPHA1': alpha1file,
+                         'ALPHA2': alpha2file,
+                         'ALPHA3': alpha3file,
+                         'CCM': CCMfile}
+
+        # Make filenames into full paths.
+        if modeldir is not None:
+            for k in names_or_objs:
+                v = names_or_objs[k]
+                if (v is not None and isinstance(v, str)):
+                    names_or_objs[k] = os.path.join(modeldir, v)
+
+        for i, key in enumerate(self.M_keys):
+            phase, wave, values = read_griddata_ascii(names_or_objs[key])
+            self._model[key] = BicubicInterpolator(phase, wave, values)
+            if key == 'M0':
+                # The "native" phases and wavelengths of the model are those
+                self._phase = np.linspace(-12., 48, 21)
+                self._wave = wave
+
+    def _flux(self, phase, wave):
+        mag_sugar = self._model['M0'](phase, wave)
+        for i, key in enumerate(self.M_keys):
+            if key != 'M0':
+                mag_sugar += self._model[key](phase, wave) * self._parameters[i]
+
+        mag_sugar += 48.59 # mag AB used in the training of SUGAR.
+        wave_factor = (wave ** 2 / 299792458. * 1.e-10)
+        return (self._parameters[0] * 10. ** (-0.4 * mag_sugar) /  wave_factor)
+
+    def bandflux_rcov(self, band, phase):
+        """
+        Model error is comming.
+        """
+        return np.zeros(phase.shape, dtype=np.float64)
```

Comment:
I think it would be best to just remove this method until it is actually implemented.

Reply:
It was removed by Florian.